### ISSUE TITLE
Docker image for arm

### ISSUE THREAD

**Issue by ghost** [HaveAGitGat/Tdarr#110]
Hello,

Can you release a docker image for ARM architecture ? I would love to get Tdarr on a Raspberry Pi

Thanks,

**Comment by martinkeat** [comment]
That'd be an interesting one. Low power and just working away in the background. I like this idea 

+1

**Comment by HaveAGitGat** [comment]
@r3vv 

This is outside of my skillset but I think it could be done with a fair amount of work. My main concern is that ARM devices are typically low powered but Tdarr is quite a resource intensive application being a transcoding focused application and all. Not sure how much demand there would be.

**Comment by HaveAGitGat** [comment]
As mentioned here: https://github.com/HaveAGitGat/Tdarr/issues/175

If someone mentions a 'much to be used' practical case for Tdarr on a Raspberry Pi I'll re-open as I think there'd be very little demand for this.

**Comment by Floppy** [comment]
I think the use case is exactly as @martinkeat points out, the ability to just work slowly away in the background. I run Plex on a pi 4, and it transcodes reasonably well on demand. Having tdarr on there to use up the spare processing power to slowly optimise my library would be amazing. But I'm not sure that counts as a "much to be used" case..?

**Comment by haragog** [comment]
I expect demand to rise significantly for either a native arm MacOs app or a docker image with arm arch, due to the new Mac mini used as "server". Curious to see the performance of Tdarr x86 app with Rosetta 2 on the new SoCs.

**Comment by sifotes** [comment]
The release of the Pi4 has greatly increased the availability of "always on" compute as well. I've been meaning to give this ticket a try but for the time being I'll watch this ticket.

**Comment by HaveAGitGat** [comment]
I now have a Raspberry Pi so will have a go at this once I’ve finished the Tdarr re-write which is nearing a useable state

**Comment by HaveAGitGat** [comment]
Spot which distributed node is the raspberry pi 4 :). The speed... think it might melt.

![image](https://user-images.githubusercontent.com/43864057/101385533-38083080-38bc-11eb-92a6-383c600a63b8.png)

**Comment by HaveAGitGat** [comment]
Tdarr v1.99.09 (v2 preview) has Tdarr_Server, Tdarr_WebUI and Tdarr_Node modules for arm (+ macos). More info here:
https://github.com/HaveAGitGat/Tdarr/blob/master/updater/README.md

![image](https://user-images.githubusercontent.com/43864057/103400576-eadb4f80-4b45-11eb-91d7-7e81281e0242.png)

**Comment by HaveAGitGat** [comment]
Some preview arm containers are now available using the .10 pre-release, been working fine on my Pi for the last few days.

![image](https://user-images.githubusercontent.com/43864057/115105000-890bed80-9f5c-11eb-9ac8-f2ade118e75a.png)

haveagitgat/tdarr_arm:2.00.10
haveagitgat/tdarr_node_arm:2.00.10

Configs are the same as non-arm (except no hardware/GPU transcoding).
https://tdarr.io/docs/installation

Will update the documentation and create a 'latest' tag next week. Let me know if you have any issues.

**Comment by marnunez** [comment]
Thanks @HaveAGitGat ! I'll give it a try.

**Comment by 2fst4u** [comment]
> Some preview arm containers are now available using the .10 pre-release, been working fine on my Pi for the last few days.
> 
> ![image](https://user-images.githubusercontent.com/43864057/115105000-890bed80-9f5c-11eb-9ac8-f2ade118e75a.png)
> 
> haveagitgat/tdarr_arm:2.00.10
> haveagitgat/tdarr_node_arm:2.00.10
> 
> Configs are the same as non-arm (except no hardware/GPU transcoding).
> https://tdarr.io/docs/installation
> 
> Will update the documentation and create a 'latest' tag next week. Let me know if you have any issues.

Hi there.

I gave this a try on a raspberry pi 4 kubernetes cluster and I am getting the error `standard_init_linux.go:219: exec user process caused: exec format error` on the server image, which to my understanding indicates incorrect architecture? I am definitely using the `haveagitgat/tdarr_arm:2.00.10` image and tag which says it's arm. My Raspberry Pis are arm64 but I wouldn't think this would be an issue would it?

**Comment by monkeyslayer56** [comment]
> > Some preview arm containers are now available using the .10 pre-release, been working fine on my Pi for the last few days.
> > ![image](https://user-images.githubusercontent.com/43864057/115105000-890bed80-9f5c-11eb-9ac8-f2ade118e75a.png)
> > haveagitgat/tdarr_arm:2.00.10
> > haveagitgat/tdarr_node_arm:2.00.10
> > Configs are the same as non-arm (except no hardware/GPU transcoding).
> > https://tdarr.io/docs/installation
> > Will update the documentation and create a 'latest' tag next week. Let me know if you have any issues.
> 
> Hi there.
> 
> I gave this a try on a raspberry pi 4 kubernetes cluster and I am getting the error `standard_init_linux.go:219: exec user process caused: exec format error` on the server image, which to my understanding indicates incorrect architecture? I am definitely using the `haveagitgat/tdarr_arm:2.00.10` image and tag which says it's arm. My Raspberry Pis are arm64 but I wouldn't think this would be an issue would it?

Having the same issue when trying to run on k3s on a Raspberry Pi 4 with aarch64.

**Comment by Just8Will** [comment]
I'm having the same issue. This error is due to the container having been set up to run for armv7 (32bit) and us using armv8+ (aarch64). The same error is given by other containers if you chose the wrong architecture tag.
I envisage not running the node at all on the raspberry pi (just the server). Then as and when my other computers come online they are running the node software and do the transcodes. Is that feasible? If not some kind of throttling / low priority mode would be super useful for the raspberry pi.

**Comment by 2fst4u** [comment]
Is aarch64 not compatible with 32bit containers? That's a shame.

**Comment by wsegatto** [comment]
Bump.

**Comment by wsegatto** [comment]
@2fst4u, requesting ARMv8 via #459.

**Comment by HaveAGitGat** [comment]
Linux_arm64 (arm v8) packages and containers are now available in 2.00.12:

https://tdarr.io/docs/installation/windows-linux-x64-arm-and-macos

```
  "2.00.12": {
    "darwin_x64": {
      "Tdarr_Node": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/darwin_x64/Tdarr_Node.zip",
      "Tdarr_Server": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/darwin_x64/Tdarr_Server.zip"
    },
    "linux_arm": {
      "Tdarr_Node": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_arm/Tdarr_Node.zip",
      "Tdarr_Server": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_arm/Tdarr_Server.zip"
    },
    "linux_arm64": {
      "Tdarr_Node": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_arm64/Tdarr_Node.zip",
      "Tdarr_Server": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_arm64/Tdarr_Server.zip"
    },
    "linux_x64": {
      "Tdarr_Node": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_x64/Tdarr_Node.zip",
      "Tdarr_Server": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_x64/Tdarr_Server.zip",
      "Tdarr_WebUI": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/linux_x64/Tdarr_WebUI.zip"
    },
    "win32_x64": {
      "Tdarr_Node": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/win32_x64/Tdarr_Node.zip",
      "Tdarr_Server": "https://f000.backblazeb2.com/file/tdarrs/versions/2.00.12/win32_x64/Tdarr_Server.zip"
    }
  }

```
https://hub.docker.com/layers/haveagitgat/tdarr_node/latest/images/sha256-a6a21d379bd2585f39bf10f62afaf4e48fe3f13553438e009344b22463d4e3dd?context=explore

https://hub.docker.com/layers/haveagitgat/tdarr_node/latest/images/sha256-a6a21d379bd2585f39bf10f62afaf4e48fe3f13553438e009344b22463d4e3dd?context=explore

**Comment by WarHawk8080** [comment]
Got it running on a DeskPiPro, RPi4 8gb, chopping along at a whopping 1 fps, but it is doing it!
Had to do a network share from my UnRaid server, the tv, movies, and encode directories, then use /etc/fstab to mount them
